src/core/adaptive/adaptive_representation_selector.ts

```diff
@@ -36,6 +33,7 @@ import SharedReference from "../../utils/reference";
 import type { CancellationSignal } from "../../utils/task_canceller";
 import TaskCanceller from "../../utils/task_canceller";
 import type { IBufferType } from "../segment_sinks";
+import type { IAdaptationStreamPlaybackObservation } from "../stream/adaptation";
```

Comment:
Isn't it weird that the ABR rely on a type defined inside another module it does not care about?

Maybe we should put that type in a more neutral place